specs/src/specs/data_square_layout.md

```diff
@@ -25,28 +25,40 @@ First, we impose some ground rules:
 
 To that end, we impose some additional rules onto _blobs only_: blobs must be placed is a way such that both the transaction sender and the block producer can be held accountable—a necessary property for e.g. fee burning. Accountable in this context means that
 
 1. The transaction sender must pay sufficient fees for blob inclusion.
 1. The block proposer cannot claim that a blob was included when it was not (which implies that a transaction and the blob it pays for must be included in the same block). In addition all blobs must be accompanied by a PayForBlob transaction.
 
-Specifically, blobs must begin at a new share. We note a nice property from this rule: if the transaction sender knows 1) `k`, the size of the matrix, 2) the starting location of their blob in a row, and 3) the length of the blob (they know this since they are sending the blob), then they can actually compute a sequence of roots to _subtrees in the row NMTs_. More importantly, anyone can compute this, and can compute _the simple Merkle root of these subtree roots_.
+Specifically, a `MsgPayForBlobs` must include a `ShareCommitment` over the contents of each blob it is paying for. If the transaction sender knows 1) `k`, the size of the matrix, 2) the starting location of their blob in a row, and 3) the length of the blob (they know this since they are sending the blob), then they can actually compute a sequence of roots to _subtrees in the row NMTs_. Taking _the simple Merkle root of these subtree roots_ provides us with the `ShareCommitment` that gets included in `MsgPayForBlobs`. Using subtree roots instead of all the leafs makes blob inclusion proofs smaller.
 
-This, however, requires the block producer to interact with the transaction sender to provide them the starting location of their blob, so that the sender can sign over the commitment based on that starting location. This can be done selectively, but is not ideal as a default for e.g. end-user wallets.
+![](./figures/subtree_roots.svg)
 
-### Blob Share Commitment Rules
+Understanding 1) and 2) would usually require interaction with the block proposer. To make the possible starting locations of blobs sufficiently predictable and to make `ShareCommitment` independent of `k`, we impose an additional rule. The blob must start at an index that is either:
 
-To make the possible starting locations of blobs sufficiently predictable and constrained such that users can deterministically compute subtree roots, needed for the `ShareCommitment`s within a PFB, without the need to interact with the block proposer, we impose one additional rule:
+1. A multiple of the `SubtreeWidth`. 
+2. A multiple of `k`, the size of the square. 
```

Comment:
[blocking] I don't think the version of non-interactive default rules that are implemented cares about the square size. If it did, then interaction would be needed between the block proposer and the msg sender.

I just updated ADR-013 w/ clarifying details but I think this meant to refer to the minimum square size that could contain the blob (a.k.a `MinSquareSize(blob)`)

Reply:
In [`NextShareIndex`](https://github.com/celestiaorg/celestia-app/blob/42433f34bce35741bb68e4ca29e3fc33e523211b/pkg/shares/blob_share_commitment_rules.go#LL51C13-L51C13) we pick the next spot as the minimum between the start of the next row and the current cursor + the `SubTreeWidth`

The `MinSquareSize(blob)` is something else but within `SubtreeWidth` function. Perhaps it does make sense to add it.

Reply:
Looking at `NextShareIndex` I'm a little confused now about if what it's doing makes sense